packages/lib/resultShareUrl/service.ts

```diff
@@ -0,0 +1,126 @@
+import { Prisma } from "@prisma/client";
+import { customAlphabet } from "nanoid";
+
+import { prisma } from "@formbricks/database";
+import { ZId } from "@formbricks/types/environment";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/types/errors";
+
+import { validateInputs } from "../utils/validate";
+
+// Create the short url and return it
+export const createResultShareKey = async (surveyId: string): Promise<string> => {
```

Comment:
Please use the `updateSurvey` function instead of this since we are updating the survey. Creating the id can move into a server action.